File: src/jkorn/pvpcore/commands/PvPCoreCommand.php
<?php

declare(strict_types=1);

namespace jkorn\pvpcore\commands;


use jkorn\pvpcore\forms\PvPCoreForms;
use jkorn\pvpcore\utils\Utils;
use pocketmine\command\Command;
use pocketmine\command\CommandSender;
use pocketmine\Player;
use pocketmine\utils\TextFormat;

class PvPCoreCommand extends Command
{

    public function __construct()
    {
        parent::__construct("pvpcoremenu", "Displays the PvPCore Menu to the sender.", "Usage: /pvpcoremenu", ["kbmenu", "pvpmenu"]);
        parent::setPermission("pvpcore.permission.menu");
    }

    /**
     * @param CommandSender $sender
     * @param string $commandLabel
     * @param string[] $args
     *
     * @return mixed
     *
     * Executes the command.
     */
    public function execute(CommandSender $sender, string $commandLabel, array $args)
    {
        if(!$sender instanceof Player)
        {
            $sender->sendMessage(Utils::getPrefix() . TextFormat::RED . " Console can't use this command.");
            return true;
        }

        if(!$this->testPermission($sender))
        {
            return true;
        }

        $form = PvPCoreForms::getPvPCoreMenu($sender);
        $sender->sendForm($form);
        return true;
    }
}

File: Nukkit/src/pvpcore/commands/PvPCoreCommand.java
package pvpcore.commands;

import cn.nukkit.Player;
import cn.nukkit.command.Command;
import cn.nukkit.command.CommandSender;
import cn.nukkit.form.window.FormWindow;
import cn.nukkit.utils.TextFormat;
import pvpcore.forms.PvPCoreForms;
import pvpcore.utils.Utils;

public class PvPCoreCommand extends Command {

    /**
     * The PvPCore Command constructor.
     */
    public PvPCoreCommand()
    {
        super("pvpcore", "Displays the PvPCore Menu to the sender.", "Usage: /pvpcore", new String[]{"pvpcoremenu", "kbmenu", "pvpmenu"});
        super.setPermission("pvpcore.permission.edit");
    }

    /**
     * Executes the command.
     * @param commandSender - The command sender.
     * @param s - The command label.
     * @param strings - The arguments for the commands.
     * @return true if command executes successfully, false otherwise.
     */
    @Override
    public boolean execute(CommandSender commandSender, String s, String[] strings)
    {
        if(!(commandSender instanceof Player))
        {
            commandSender.sendMessage(Utils.getPrefix() + TextFormat.RED.toString() + " Console can't use this command.");
            return true;
        }

        if(!this.testPermission(commandSender))
        {
            return true;
        }

        FormWindow window = PvPCoreForms.getPvPCoreMenu((Player)commandSender);
        ((Player) commandSender).showFormWindow(window);
        return true;
    }
}
